pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -405,80 +410,100 @@ func (f *fqdnController) deleteRuleSelectedPods(ruleID string) error {
 
 func (f *fqdnController) onDNSResponse(
 	fqdn string,
-	responseIPs map[string]net.IP,
-	lowestTTL uint32,
+	newDNSresponseIPs map[string]ipWithTTL,
 	waitCh chan error,
 ) {
-	if len(responseIPs) == 0 {
+	if len(newDNSresponseIPs) == 0 {
 		klog.V(4).InfoS("FQDN was not resolved to any addresses, skip updating DNS cache", "fqdn", fqdn)
 		if waitCh != nil {
 			waitCh <- nil
 		}
 		return
 	}
-	// mustCacheResponse is only true if the FQDN is already tracked by this
-	// controller, or it matches at least one fqdnSelectorItem from the policy rules.
-	// addressUpdate is only true if there has been an update in IP addresses
-	// corresponded with the FQDN.
-	mustCacheResponse, addressUpdate := false, false
-	recordTTL := time.Now().Add(time.Duration(lowestTTL) * time.Second)
+
+	currentTime := time.Now()
+	ipWithTTLMap := make(map[string]ipWithTTL)
+	minTimeToReQuery := time.Unix(1<<63-62135596801, 999999999)
+	addressUpdate := false
+
+	ipMapFiller := func(ip string, ipMeta ipWithTTL, minTime *time.Time) {
```

Comment:
the name of this function (`ipMapFiller`) is really hurting the readability of the code. I would suggest something like `addIPtoMap` / `addIPToCache` / `addIPToNewCache` / `setIPInMap` / ...

Additionally, `minTime` does not need to be a parameter to this function: you are always using `&minTimeToReQuery` as the argument value, and `minTimeToReQuery` is captured by the closure (by reference).

Reply:
Yes , my bad. The closure captures the variable ```minTimeToReQuery``` and should not be passed as an argument.